paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -394,49 +389,35 @@ internal class DefaultFlowController @Inject internal constructor(
         }
     }
 
-    internal fun onPaymentFlowResult(
-        paymentFlowResult: PaymentFlowResult.Unvalidated
-    ) {
+    internal fun onPaymentResult(paymentResult: PaymentResult) {
         lifecycleScope.launch {
-            runCatching {
-                paymentFlowResultProcessorProvider.get().processResult(
-                    paymentFlowResult
-                )
-            }.fold(
-                onSuccess = {
-                    withContext(uiContext) {
-                        paymentResultCallback.onPaymentSheetResult(
-                            createPaymentSheetResult(it)
-                        )
-                    }
-                },
-                onFailure = {
-                    withContext(uiContext) {
-                        paymentResultCallback.onPaymentSheetResult(
-                            PaymentSheetResult.Failed(it)
-                        )
-                    }
+            paymentResultCallback.onPaymentSheetResult(
+                withContext(uiContext) {
+                    createPaymentSheetResult(paymentResult)
                 }
             )
         }
     }
 
     private fun createPaymentSheetResult(
-        stripeIntentResult: StripeIntentResult<StripeIntent>
-    ) = when (stripeIntentResult.outcome) {
-        StripeIntentResult.Outcome.SUCCEEDED -> {
+        paymentResult: PaymentResult
+    ): PaymentSheetResult = when (paymentResult) {
+        is PaymentResult.Completed -> {
             PaymentSheetResult.Completed
         }
-        StripeIntentResult.Outcome.CANCELED -> {
+        is PaymentResult.Canceled -> {
             PaymentSheetResult.Canceled
         }
+        is PaymentResult.Failed -> {
+            PaymentSheetResult.Failed(
+                IllegalArgumentException(
+                    "Failed to confirm intent: ${paymentResult.throwable.message}"
```

Comment:
I assume this came from the lastErrorMessage on the intent?

Reply:
Correct. The StripeIntent error message is piggybacked onto the error here.